trax/tf_numpy/examples/mnist/model.py

```diff
@@ -19,119 +19,109 @@
+    the hidden layers are specified by the user in `hidden_layers` and the
+    network takes care of adding layers to match the input and output size.
 
-  Attributes:
+    Attributes:
     weights: A list of 2-d float32 arrays containing the layer weights.
     biases: A list of 2-d float32 arrays containing the layer biases.
 
-  Methods:
+    Methods:
     forward: Can be used to perform a forward pass on a batch of
       flattened images. Output is returned as a batch of one-hot vectors of the
       classes.
     train: method performs a forward and backward pass and updates the
       weights and biases.
     evaluate: method can be used to evaluate the network on a batch of
       examples.
-  """
-
-  def __init__(self, hidden_layers, input_size=784, num_classes=10):
-    """Initializes the neural network.
-
-    Args:
-      hidden_layers: List of ints specifying the sizes of hidden layers. Could
-        be empty.
-      input_size: Length of the input array. The network receives the input
-        image as a flattened 1-d array. Defaults to 784(28*28), the default
-        image size for MNIST.
-      num_classes: The number of output classes. Defaults to 10.
-    """
-    hidden_layers = [input_size] + hidden_layers + [num_classes]
-    self.weights = []
-    self.biases = []
-    for i in range(len(hidden_layers) - 1):
-      # TODO(srbs): This is manually cast to float32 to avoid the cast in
-      # np.dot since backprop fails for tf.cast op.
-      self.weights.append(
-          np.array(
-              np.random.randn(hidden_layers[i + 1], hidden_layers[i]),
-              copy=False,
-              dtype=float32))
-      self.biases.append(
-          np.array(
-              np.random.randn(hidden_layers[i + 1]), copy=False, dtype=float32))
-
-  def forward(self, x):
-    """Performs the forward pass.
-
-    Args:
-      x: 2-d array of size batch_size x image_size.
-
-    Returns:
-      A 2-d array of size batch_size x num_classes.
     """
 
-    def sigmoid(x):
-      return 1.0 / (1.0 + np.exp(-x))
+    def __init__(self, hidden_layers, input_size=784, num_classes=10, learning_rate=0.001):
+        """Initializes the neural network.
 
-    for w, b in zip(self.weights, self.biases):
-      x = sigmoid(np.dot(w, x.T).T + b)
-    return x
+        Args:
+          hidden_layers: List of ints specifying the sizes of hidden layers. Could
+            be empty.
+          input_size: Length of the input array. The network receives the input
+            image as a flattened 1-d array. Defaults to 784(28*28), the default
+            image size for MNIST.
+          num_classes: The number of output classes. Defaults to 10.
+        """
+        hidden_layers = [input_size] + hidden_layers + [num_classes]
+        self.params = []
+        for i in range(len(hidden_layers) - 1):
+            self.params.append(np.random.randn(hidden_layers[i + 1], hidden_layers[i]))
```

Comment:
How about a little more structure here?

Maybe the params could be a list of Namedtuples or even plain tuples? 

Layer = collections.namedtuple('Layer', ['weights', 'biases'])?